src/Controllers/ContentController.php

```diff
@@ -0,0 +1,651 @@
+     */
+    private $stripTagDecorator;
+
+    /**
+     * ContentController constructor.
+     *
+     * @param ContentService $contentService
+     * @param CommentService $commentService
+     * @param VimeoVideoSourcesDecorator $vimeoVideoDecorator
+     * @param UserProviderInterface $userProvider
+     * @param MailService $mailoraMailService
+     * @param ContentHierarchyRepository $contentHierarchyRepository
+     * @param FullTextSearchService $fullTextSearchService
+     * @param StripTagDecorator $stripTagDecorator
+     */
+    public function __construct(
+        ContentService $contentService,
+        CommentService $commentService,
+        VimeoVideoSourcesDecorator $vimeoVideoDecorator,
+        UserProviderInterface $userProvider,
+        MailService $mailoraMailService,
+        ContentHierarchyRepository $contentHierarchyRepository,
+        FullTextSearchService $fullTextSearchService,
+        StripTagDecorator $stripTagDecorator
+    ) {
+        $this->contentService = $contentService;
+        $this->commentService = $commentService;
+        $this->vimeoVideoDecorator = $vimeoVideoDecorator;
+        $this->userProvider = $userProvider;
+        $this->mailoraMailService = $mailoraMailService;
+        $this->contentHierarchyRepository = $contentHierarchyRepository;
+        $this->fullTextSearchService = $fullTextSearchService;
+        $this->stripTagDecorator = $stripTagDecorator;
+    }
+
+    /**
+     * @param $contentId
+     * @param Request $request
+     * @return JsonResponse
+     * @throws NonUniqueResultException
+     */
+    public function getContent($contentId, Request $request)
+    {
+        $content = $this->contentService->getById($contentId);
+
+        if (!$content) {
+            return response()->json();
+        }
+
+        $isDownload = $request->get('download', false);
+
+        $parent = array_first(
+            $this->contentService->getByChildIdWhereParentTypeIn(
+                $contentId,
+                ['course', 'song', 'learning-path', 'pack', 'pack-bundle']
+            )
+        );
+
+        $parentChildren = $parent['lessons'] ?? $this->contentService->getFiltered(
+                $request->get('page', 1),
+                $request->get('limit', 10),
+                '-published_on',
+                [$content['type']],
+                [],
+                [],
+                [],
+                [],
+                [],
+                [],
+                false,
+                false,
+                false
+            )['results'];
+
+        ModeDecoratorBase::$decorationMode = DecoratorInterface::DECORATION_MODE_MINIMUM;
+
+        $parentChildrenTrimmed = $this->getParentChildTrimmed($parentChildren, $content);
+
+        $content['related_lessons'] = $parentChildrenTrimmed;
+
+        $content = $this->attachNextPrevLesson($parent, $content, $parentChildren);
+
+        CommentRepository::$availableContentId = $content['id'];
```

Comment:
Some of these controller methods could use some better spacing and some comments to explain what is going on. I had a hard time following these functions.